Libraries/LibGUI/Window.cpp

```diff
@@ -612,11 +614,11 @@ void Window::save_to(AK::JsonObject& json)
 
 void Window::set_fullscreen(bool fullscreen)
 {
+    if (!is_visible())
+        return;
     if (m_fullscreen == fullscreen)
         return;
     m_fullscreen = fullscreen;
-    if (!m_window_id)
-        return;
```

Comment:
This is behavior-breaking. It should be possible to set up a fullscreen window and mark it as fullscreen before calling `show()`.

Reply:
@awesomekling you're right! Moved the visibility check after toggling the full screen state.